### PR TITLE
[ENH]: Retrieve core list from resen-core repo

### PR DESCRIPTION
**Pull request by ljlamarche**
This PR modifies how resen gets its list of valid cores.  Instead of the hard-coded list, resen now downloads [cores.json](https://github.com/EarthCubeInGeo/resen-core/blob/master/cores.json) from the [resen-core](https://github.com/EarthCubeInGeo/resen-core) repo if there isn't already a list of cores available.  Lists of cores are contained in `~/.config/resen/cores`.  All files in this directory will be read and appended together to form the core list, so this technically sets up structure for users to add their own cores by adding their own file (say `my_cores.json`) to `~/.config/resen/cores`.  Additionally, this PR adds the `update` command, which lets users manually update the core list if we release a new core without reinstalling resen.  This PR should take care of some of the background work required for #64.

### CODE REVIEW

**Comment by asreimer** [comment]
This PR wasn't tracking the current develop branch, so I merged that in.

File: resen/Resen.py [hunk 1]
```diff
@@ -832,19 +834,32 @@ def update_bucket_statuses(self):
...
+        core_dir = os.path.join(self.resen_root_dir,'cores')
+
+        # If core directory does not exist, create it and download the default core list file
+        if not os.path.exists(core_dir):
+            os.makedirs(core_dir)
+            self.update_core_list()
+
+        # for each JSON file in core directory, read in list of cores
+        cores = []
+        for fn in os.listdir(core_dir):
```
Comment (asreimer) [inline resen/Resen.py:858]: Not a big deal right now, but this could have side effects:
1) doesn't check validity of JSON (do dictionaries have `version`, `image_id`, etc?)
2) could fail if problem reading files

Something to keep in mind.

File: setup.py [hunk 2]
```diff
@@ -10,13 +10,16 @@
 # To use a consistent encoding
 from codecs import open
 from os import path
+import os
```
Comment (asreimer) [inline setup.py]: Why is this here?

**Review by asreimer — APPROVED** [review]

File: resen/Resen.py (hunk 1 above)
Comment (ljlamarche) [reply resen/Resen.py:858]: Good point, particularly if we're going to have users add their own JSON files.  I've added exception handling if the JSON parsing fails (not a valid JSON file), but maybe we can hold off on checking that the dictionaries are correct until we decide exactly how the user interface for adding new cores is going to work (next release).

File: setup.py (hunk 2 above)
Comment (ljlamarche) [reply setup.py]: Residual.  It's been removed.
